packages/core/src/Cardano/types/ProtocolParameters.ts

```diff
@@ -24,27 +27,87 @@ export interface ValidityInterval {
   invalidHereafter?: Slot;
 }
 
-export interface ProtocolParametersBabbage {
-  minFeeCoefficient?: number;
-  minFeeConstant?: number;
-  maxBlockBodySize?: number;
-  maxBlockHeaderSize?: number;
-  maxTxSize?: number;
-  stakeKeyDeposit?: number;
-  poolDeposit?: number;
-  poolRetirementEpochBound?: number;
-  desiredNumberOfPools?: number;
-  poolInfluence?: number;
-  monetaryExpansion?: number;
-  treasuryExpansion?: number;
-  minPoolCost?: number;
-  coinsPerUtxoByte?: number;
-  maxValueSize?: number;
-  collateralPercentage?: number;
-  maxCollateralInputs?: number;
-  protocolVersion?: ProtocolVersion;
-  costModels?: CostModels;
-  prices?: Prices;
-  maxExecutionUnitsPerTransaction?: ExUnits;
-  maxExecutionUnitsPerBlock?: ExUnits;
+export interface TxFeePolicy {
+  coefficient: string;
+  constant: number;
 }
+
+export interface SoftforkRule {
+  initThreshold: string;
+  minThreshold: string;
+  decrementThreshold: string;
+}
+
+export type ProtocolParametersByron = {
+  heavyDlgThreshold: string;
+  maxBlockSize: number;
+  maxHeaderSize: number;
+  maxProposalSize: number;
+  maxTxSize: number;
+  mpcThreshold: string;
+  scriptVersion: number;
+  slotDuration: number;
+  unlockStakeEpoch: number;
+  updateProposalThreshold: string;
+  updateProposalTimeToLive: number;
+  updateVoteThreshold: string;
+  txFeePolicy: TxFeePolicy;
+  softforkRule: SoftforkRule;
+};
+
+type NewProtocolParamsInShelley = {
+  minFeeCoefficient: number;
+  minFeeConstant: number;
+  maxBlockBodySize: number;
+  maxBlockHeaderSize: number;
+  stakeKeyDeposit: number;
+  poolDeposit: number | null;
+  poolRetirementEpochBound: number;
+  desiredNumberOfPools: number;
+  poolInfluence: string;
+  monetaryExpansion: string;
+  treasuryExpansion: string;
+  decentralizationParameter: string;
+  minUtxoValue: number;
+  minPoolCost: number;
+  extraEntropy: 'neutral' | string;
+  protocolVersion: ProtocolVersion;
+};
+
+export type ShelleyProtocolParams = Omit<
+  ProtocolParametersByron,
+  | 'heavyDlgThreshold'
+  | 'maxBlockSize'
+  | 'maxHeaderSize'
+  | 'maxProposalSize'
+  | 'mpcThreshold'
+  | 'scriptVersion'
+  | 'slotDuration'
+  | 'unlockStakeEpoch'
+  | 'updateProposalThreshold'
+  | 'updateProposalTimeToLive'
+  | 'updateVoteThreshold'
+  | 'txFeePolicy'
+  | 'softforkRule'
+> &
+  NewProtocolParamsInShelley;
+
+type NewProtocolParamsInAlonzo = {
+  coinsPerUtxoWord: number;
+  maxValueSize: number;
+  collateralPercentage: number;
+  maxCollateralInputs: number;
+  costModels: CostModels;
+  prices: Prices;
+  maxExecutionUnitsPerTransaction: ExUnits;
+  maxExecutionUnitsPerBlock: ExUnits;
+};
+
+export type AlonzoProtocolParams = Omit<ShelleyProtocolParams, 'minUtxoValue'> & NewProtocolParamsInAlonzo;
+
+type NewProtocolParamsInBabbage = {
+  coinsPerUtxoByte: number;
+};
+
+export type ProtocolParameters = Omit<AlonzoProtocolParams, 'coinsPerUtxoWord' | 'extraEntropy'> &
```

Comment:
```suggestion
export type BabbageProtocolParameters = Omit<AlonzoProtocolParams, 'coinsPerUtxoWord' | 'extraEntropy'> &
export type ProtocolParameters = BabbageProtocolParameters;
```
Better if `ProtocolParameters` is just an alias to the current era

Reply:
Applied here 004f780300900891ba24d193125644e6c31f8d30

Reply:
The idea was to maintain the interfaces using `ProtocolParameters`, not change to `BabbageProtocolParameters`, by aliasing like this:

```ts
export type ProtocolParameters = BabbageProtocolParameters;
```

Reply:
6d49527a6a5de79d9e8350a785f02acfdf560fb0